Java/core/src/main/java/com/amazon/randomcutforest/store/PointStore.java

```diff
@@ -30,6 +30,9 @@
  */
 public abstract class PointStore<Store, Point> extends IndexManager implements IPointStore<Point> {
 
+    public static int INFEASIBLE_LOCATION = -1;
+
+    public static int INFEASIBLE_INDEX = -1;
```

Comment:
If we're going to have 2 different constants, should we assign them different values? Or maybe we only need one constant?

Reply:
Two constants (with different negative values) sound like a good idea. Using the same constant becomes problematic in sense of type checking.